File: app/src/main/java/com/adrar/adrarconnect/utils/SharedPreferenceUtils.java
package com.adrar.adrarconnect.utils;

//
// Created by FERRARIS Philippe on 23/07/2019 for adrarConnect.
//

import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;

import com.adrar.adrarconnect.data.model.AccueilBean;
import com.adrar.adrarconnect.data.staticData.Constants;

public class SharedPreferenceUtils extends AppCompatActivity {

    SharedPreferences sharedPreferences = getBaseContext().getSharedPreferences(Constants.PREFS, MODE_PRIVATE);

    public void saveAccueil(AccueilBean a) {
        sharedPreferences
                .edit()
                .putString(Constants.PREFS_ACCUEIL, a + "")
                .apply();
    }
}

File: app/src/main/java/com/adrar/adrarconnect/utils/sharedPreferenceUtils.kt
package com.adrar.adrarconnect.utils

import android.content.Context.MODE_PRIVATE
import com.adrar.adrarconnect.data.model.AccueilBean
import com.adrar.adrarconnect.data.model.LoginBean
import com.adrar.adrarconnect.data.staticData.Constants

private const val SPFileName = "SPFileName.xml"


private const val SAVE_ACCUEIL_KEY = "SAVE_ACCUEIL_KEY"

//SharedPreferenceUtilsKt.saveAcceuil(accueilBean);
fun saveAcceuil(data: AccueilBean) =
        MyApplication.context.getSharedPreferences(SPFileName, MODE_PRIVATE).edit().putString(SAVE_ACCUEIL_KEY, Constants.GSON.toJson(data)).apply()


//AccueilBean accueilBean = SharedPreferenceUtilsKt.getAccueil();
fun getAccueil(): AccueilBean? {

    val json: String? = MyApplication.context.getSharedPreferences(SPFileName, MODE_PRIVATE).getString(SAVE_ACCUEIL_KEY, null)

    return if (json != null) {
        Constants.GSON.fromJson(json, AccueilBean::class.java)
    } else {
        null
    }
}

private const val SAVE_LOGINBEAN = "SAVE_LOGINBEAN"

//SharedPreferenceUtilsKt.saveUser(LoginBean);
fun saveUser(data: LoginBean) =
        MyApplication.context.getSharedPreferences(SPFileName, MODE_PRIVATE).edit().putString(SAVE_LOGINBEAN, Constants.GSON.toJson(data)).apply()

//LoginBean loginBean = SharedPreferenceUtilsKt.getUser();
fun getUser(): LoginBean? {

    val json: String? = MyApplication.context.getSharedPreferences(SPFileName, MODE_PRIVATE).getString(SAVE_LOGINBEAN, null)

    return if (json != null) {
        Constants.GSON.fromJson(json, LoginBean::class.java)
    } else {
        null
    }
}